### ISSUE TITLE
ETC Echo Integration Interface

### ISSUE THREAD

**Issue by wdcflash** [bitfocus/companion-module-requests#640]
I would like to add the companion integration with the ETC Echo Integration Interface. It used UDP to receive triggers and even gives a UDP response. We have gotten it working with generic UDP Instance and would like to add the ability to get feedback also. Is it possible for me to be able to program this out to submit it to be added to companion? 

[Echo_IntegrationInterface_InstallGuide_revA.pdf](https://github.com/bitfocus/companion-module-requests/files/7565846/Echo_IntegrationInterface_InstallGuide_revA.pdf)

**Comment by josephdadams** [comment]
Sure. Here is the developers guide. Once you are ready for a repo to be made, tag me here. 

https://github.com/bitfocus/companion/wiki/Module-Development

**Comment by zklosko** [comment]
I've started working on one [here ](https://github.com/zklosko/companion-module-etc-echo)if anyone would like to help. Will be testing it on Monday.

**Comment by abf-cthornton** [comment]
> I've started working on one [here ](https://github.com/zklosko/companion-module-etc-echo)if anyone would like to help. Will be testing it on Monday.

I was able to get this installed on my CompanionPI, and even see it in the list of modules, but I am getting an error of "No config data loaded".  I know enough to be dangerous here so I have checked permissions and ownership against other directories that exist but still get the same error.

<img width="562" height="665" alt="Image" src="https://github.com/user-attachments/assets/89a199ec-2027-458d-8ed0-3a5640a286b2" />

**Comment by peternewman** [comment]
What does the module's log say, under the >_ menu? Have you installed the module dependencies (in this case by doing a `yarn install`)?

**Comment by zklosko** [comment]
@abf-cthornton thanks for pointing that out. Can you please open an issue [here](https://github.com/zklosko/companion-module-etc-echo/issues) for me to look at?

**Comment by abf-cthornton** [comment]
> [@abf-cthornton](https://github.com/abf-cthornton) thanks for pointing that out. Can you please open an issue [here](https://github.com/zklosko/companion-module-etc-echo/issues) for me to look at?

Done!

**Comment by abf-cthornton** [comment]
> What does the module's log say, under the >_ menu? Have you installed the module dependencies (in this case by doing a `yarn install`)?

This is the only thing the log shows, even after turning the module off and on again:
System: Starting log. Only lines generated since opening the page are shown here

yarn command is not found on my raspberry pi. I looked into it and npm command is not found either. Ran out of time today to keep trying today.

**Comment by zklosko** [comment]
If anyone is interesting in helping me test, I have a mostly-completed working module here: https://github.com/zklosko/companion-module-etc-echo/releases/tag/v0.2.1

**Comment by peternewman** [comment]
> If anyone is interesting in helping me test, I have a mostly-completed working module here: https://github.com/zklosko/companion-module-etc-echo/releases/tag/v0.2.1

It's probably worth you starting the repository request process in parallel @zklosko . You can ask for one in the module-development channel.

**Comment by josephdadams** [comment]
This is now available as the **`etc-echo`** module (latest version v1.0.1).

You can install it from Companion's module store, or see https://bitfocus.io/connections/etc-echo

Closing as implemented — please open a new issue on [the module's repository](https://github.com/bitfocus/companion-module-etc-echo/issues) if something specific is missing.
